### ISSUE TITLE
NX 14.2.4 incompatibility

### ISSUE THREAD

**Issue by radulle** [bikecoders/ngx-deploy-npm#296]
After upgrade to NX 14.2.4 getting:

```
Cannot find module 'nx/src/utils/app-root'
Require stack:
- /Users/nradulaski/code/hub/node_modules/ngx-deploy-npm/node_modules/@nrwl/devkit/index.js
- /Users/nradulaski/code/hub/node_modules/ngx-deploy-npm/src/executors/deploy/executor.js
- /Users/nradulaski/code/hub/node_modules/nx/src/config/workspaces.js
- /Users/nradulaski/code/hub/node_modules/nx/src/command-line/run.js
- /Users/nradulaski/code/hub/node_modules/nx/bin/run-executor.js
```

In NX `app-root` was renamed to `workspace-root`. `@nrwl/devkit` is locked to 14.0.5.

Having it ^14.0.5 should fix the issue.

**Comment by radulle** [comment]
This [PR](https://github.com/bikecoders/ngx-deploy-npm/pull/297) should fix it.

**Comment by dianjuar** [comment]
Thank you all for providing your feedback
Thank you for your patience and initiative on this issue.

We got blocked on some deployment issues with version V4.0.0. This block was the reason for our delay in providing the fix. We could solve the case, and this fix is available under V4.0.1 🎉 

The breaking change introduced on V4 is simply following with the [end of life of nodeV12](https://endoflife.date/nodejs). We are not longer going to offer support for node V12. You can update to V4 without problems if your projects don't run on node12.

**Comment by kurtinatlanta** [comment]
Thank you!
